NuRadioReco/modules/channelSinewaveSubtraction.py

```diff
@@ -0,0 +1,404 @@
+from NuRadioReco.modules.base.module import register_run
+from NuRadioReco.utilities import units, fft
+
+import matplotlib.pyplot as plt
+from scipy.optimize import curve_fit
+from scipy.signal import lfilter
+import numpy as np
+import sys
+import time
+import logging
+logger = logging.getLogger("NuRadioReco.modules.channelSinewaveSubtraction")
+
+"""
+This module provides a class for continuous wave (CW) noise filtering using sine subtraction.
+In contrast to the module channelCWNOtchFilter, which uses a notch filter to remove CW noise.
+"""
+
+
+class channelSinewaveSubtraction:
+    """ Continuous wave (CW) filter module. Uses sine subtraction based on scipy curve_fit. """
+    def __init__(self):
+        pass
+
+    def begin(self, save_filtered_freqs: bool = False, freq_band: tuple[float, float] = (0.1, 0.7)) -> None:
+        """
+        Initialize the CW filter module.
+
+        Parameters
+        ----------
+        save_filtered_freqs: bool (default: False)
+            Flag to save the identified noise frequencies for each channel.
+        freq_band: tuple (default: (0.1, 0.7))
+            Frequency band to calculate baseline RMS of fft spectrum. Used to identify noise peaks.
+            0.1 to 0.7 GHz is the default for RNO-G, based on bandpass.
+
+        """
+        self.save_filtered_freqs = [] if save_filtered_freqs else None
+        self.freq_band = freq_band
+
+    @register_run()
+    def run(self, event: 'NuRadioReco.framework.event.Event', station: 'NuRadioReco.framework.station.Station', det: 'NuRadioReco.detector.Detector' = None, peak_prominence: float = 4.0) -> None:
+        """
+        Run the CW filter module on a given event and station. Removes all the CW peaks > peak_prominence * RMS.
+
+        Parameters
+        ----------
+        event: `NuRadioReco.framework.event.Event`
+            Event object to process.
+        station: `NuRadioReco.framework.station.Station`
+            Station object to process.
+        det: `NuRadioReco.detector.detector.Detector` (default: None)
```

Comment:
I think the documentation currently fails because of this. `NuRadioReco.detector.detector.Detector` is a factory function and probably not suited to specify a type.